src/entity/RDFPropertyMap.cpp

```diff
@@ -3,8 +3,54 @@
 
 namespace sempr { namespace entity {
 
+// triple iterator for rdfpropertymap
+
+RDFPropertyMapIterator_impl::RDFPropertyMapIterator_impl(
+    const RDFPropertyMap* pmap, std::map<std::string, RDFValue>::const_iterator it)
+    : it_(it), pmap_(pmap)
+{
+}
+
+const Triple& RDFPropertyMapIterator_impl::operator*() const
+{
+    static Triple t;
```

Comment:
Why static?
t is changed by every call. It also could be non static and non reference to get a copy of the created object.

Reply:
Because then the `operator ->() const` wouldn't work anymore.